api/src/utils/sanitize-schema.ts

```diff
@@ -0,0 +1,76 @@
+import { Field, Relation } from '@directus/shared/types';
+import { omit } from 'lodash';
+import { Collection } from '../types';
+
+/**
+ * Omit certain database vendor specific collection properties that should not be compared when performing diff
+ *
+ * @param collection collection to sanitize
+ * @returns sanitized collection
+ *
+ * @see {@link https://github.com/knex/knex-schema-inspector/blob/master/lib/types/table.ts}
+ */
+
+export function sanitizeCollection(collection: Collection | undefined) {
+	const omittedPaths = [
```

Comment:
Instead of omitting paths, should we rely on an allow-list of keys that we know we use / rely on and use `pick`?

Reply:
I think omitting path was first chosen because for Field, the omitting is "simpler" than allowing 😄 but allow-list does seem safer in case there's ever new keys added in knex-schema-inspector side I suppose. Will switch to allow-list 👍